packages/m365_defender/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,338 @@
+
+  #######################
+  ## ECS Event Mapping ##
+  #######################
+  - set:
+      field: event.kind
+      value: alert
+  # Events returned from the API is always in UTC, so should never use anything else
+  - set:
+      field: event.timezone
+      value: UTC
+  - set:
+      field: event.action
+      copy_from: json.alerts.category
+      ignore_empty_value: true
+  - set:
+      field: event.provider
+      copy_from: json.alerts.serviceSource
+      ignore_empty_value: true
+  - set:
+      field: event.created
+      copy_from: json.createdTime
+      ignore_empty_value: true
+  - append:
+      field: event.category
+      value: host
+  - append:
+      field: event.category
+      value: malware
+      if: ctx.json?.determination == 'Malware'
+  - append:
+      field: event.category
+      value: process
+      if: ctx.json?.entities?.entityType == 'Process'
+  - append:
+      field: event.type
+      value: user
+      if: ctx.json?.entities?.entityType == 'User'
+  - append:
+      field: event.type
+      value:
+        - creation
+        - start
+      if: ctx.json?.status == 'New'
+  - append:
+      field: event.type
+      value: end
+      if: ctx.json?.status == 'Resolved'
+  - rename:
+      field: json.alerts.alertId
+      target_field: event.id
+      ignore_missing: true
+  - rename:
+      field: json.alerts.firstActivity
+      target_field: event.start
+      ignore_missing: true
+  - rename:
+      field: json.alerts.lastActivity
+      target_field: event.end
+      ignore_missing: true
+  - set:
+      field: event.severity
+      value: 0
+      if: ctx.json?.severity == 'Unspecified'
+  - set:
+      field: event.severity
+      value: 1
+      if: ctx.json?.severity == 'Informational'
+  - set:
+      field: event.severity
+      value: 2
+      if: ctx.json?.severity == 'Low'
+  - set:
+      field: event.severity
+      value: 3
+      if: ctx.json?.severity == 'Medium'
+  - set:
+      field: event.severity
+      value: 4
+      if: ctx.json?.severity == 'High'
+  - script:
+      lang: painless
+      if: "ctx.event?.start != null && ctx.event?.end != null"
+      source: >
+        Instant eventstart = ZonedDateTime.parse(ctx.event?.start).toInstant();
+        Instant eventend = ZonedDateTime.parse(ctx.event?.end).toInstant();
+        ctx.event['duration'] = ChronoUnit.NANOS.between(eventstart, eventend);
+
+  ########################
+  ## ECS Threat Mapping ##
+  ########################
+  - set:
+      field: threat.framework
+      value: MITRE ATT&CK
+      if: ctx.json?.alerts?.category != null
+  - rename:
+      field: json.alerts.category
+      target_field: threat.technique.name
+      ignore_missing: true
+  - rename:
+      field: json.alerts.description
+      target_field: rule.description
+      ignore_missing: true
+      if: ctx.json?.alerts != null && ctx.json?.alerts?.description != null && ctx.json?.alerts?.description.length() < 1020
```

Some checks are redundant.
```suggestion
      if: ctx.json?.alerts?.description != null && ctx.json.alerts.description.length() < 1020
```

I wonder why rename only when the length doesn't exceed 1020 chars? Maybe we want to truncate instead.